nir/src/main/scala/scala/scalanative/nir/serialization/BinarySerializer.scala

```diff
@@ -13,7 +14,7 @@ final class BinarySerializer {
   private[this] val buffer           = new DataOutputStream(bufferUnderyling)
 
   private[this] var lastPosition: Position = Position.NoPosition
-  private[this] val fileIndexMap           = mutable.ListMap.empty[URI, Int]
+  private[this] var fileIndexMap           = ListMap.empty[URI, Int]
```

Comment:
That doesn't seem very efficient. Consider using a `mutable.ListBuffer` *and* a `mutable.HashMap` instead, like Scala.js does:
https://github.com/scala-js/scala-js/blob/a799b7ba37acf15c459c311ec69b1a292398ee7a/ir/src/main/scala/org/scalajs/ir/Serializers.scala#L120-L123